test/oracles/UniswapV3Oracle.test.ts

```diff
@@ -0,0 +1,292 @@
+import { ethers } from "hardhat";
+import { time } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+import { Reverter } from "@/test/helpers/reverter";
+import { BigNumberish } from "ethers";
+import { wei } from "@/scripts/utils/utils";
+
+import { UniswapV3OracleMock, UniswapV3FactoryMock, UniswapV3PoolMock } from "@ethers-v6";
+
+describe("UniswapV3Oracle", () => {
+  const reverter = new Reverter();
+
+  const A_TOKEN = "0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa";
+  const B_TOKEN = "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB";
+  const C_TOKEN = "0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC";
+  const A_B_PATH = [A_TOKEN, B_TOKEN];
+
+  const PERIOD = 2;
+  const enum FeeAmount {
+    LOW = 500,
+    MEDIUM = 3000,
+  }
+
+  let oracle: UniswapV3OracleMock;
+  let uniswapV3Factory: UniswapV3FactoryMock;
+
+  before("setup", async () => {
+    const UniswapV3FactoryMock = await ethers.getContractFactory("UniswapV3FactoryMock");
+    const Oracle = await ethers.getContractFactory("UniswapV3OracleMock");
+
+    uniswapV3Factory = await UniswapV3FactoryMock.deploy();
+    oracle = await Oracle.deploy();
+
+    await oracle.__OracleV3Mock_init(await uniswapV3Factory.getAddress());
+
+    await reverter.snapshot();
+  });
+
+  async function createPools(token0: string, token1: string): Promise<UniswapV3PoolMock> {
+    await uniswapV3Factory.createPool(token0, token1, FeeAmount.MEDIUM);
+
+    let poolAddress = await uniswapV3Factory.getPool(token0, token1, FeeAmount.MEDIUM);
+
+    return <UniswapV3PoolMock>await ethers.getContractAt("UniswapV3PoolMock", poolAddress);
+  }
+
+  function encodePriceSqrt(reserve1: number, reserve0: number): BigNumberish {
+    return BigInt(Math.sqrt(reserve1 / reserve0) * Math.pow(2, 96));
+  }
+
+  function getPriceByTick(tick: number, amount: BigInt): BigInt {
+    return BigInt(Math.floor(1.0001 ** tick * Number(amount)));
+  }
```

Comment:
Make these functions' inputs and outputs BigNumberish. Maybe it's also better to use bignumberjs to avoid precision inaccuracies